### PR TITLE
feat: Allow users to create custom domains to prepare launch

### LINKED ISSUE #4176: Update app UI to allow users to manage domains with site branch configs
## TODO

Review the configuration options for a site/branch and define our goals or desired outcomes.

As a user,  I want to be able to add as many branch specific configurations and custom domains to my site as possible.

## Acceptance Criteria

- [x] Update pages UI to only update configuration when the user inputs/updates the configuration for a branch
- [x] Allow users to create, edit, delete their own custom domains and branch configs before provisioning

### PR DESCRIPTION
**Pull request by apburnes**
Closes #4176 

## Changes proposed in this pull request:
- Adds API endpoints to create, update and delete pending domains
- Adds UI to custom domains to create, edit, and delete pending domains

## security considerations
None

### CODE REVIEW

File: api/controllers/domain.js [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+    if (!domain) {
+      return res.notFound();
+    }
+
+    if (domain.state !== 'pending') {
+      return res.badRequest({
+        message: `
+          The domain cannot be deleted because it is ${domain.state}.
+          Please contact to cloud.gov Pages support.
```
Comment (svenaas) [inline api/controllers/domain.js:121]:
```suggestion
          Please contact cloud.gov Pages support.
```

File: api/controllers/domain.js [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+    if (!domain) {
+      return res.notFound();
+    }
+
+    if (domain.state !== 'pending') {
+      return res.badRequest({
+        message: `
+          The domain cannot be updated because it is ${domain.state}.
+          Please contact to cloud.gov Pages support.
```
Comment (svenaas) [inline api/controllers/domain.js:178]:
```suggestion
          Please contact cloud.gov Pages support.
```

**Review by svenaas — CHANGES_REQUESTED** [review]
Couple of typos in error messages.

**Review by drewbo — CHANGES_REQUESTED** [review]
Voice discussion of controller improvements for finding matching sbc/domain name. Otherwise good

File: api/controllers/domain.js (hunk 1 above)
Comment (apburnes) [reply api/controllers/domain.js]: Updated

**Review by drewbo — APPROVED** [review]

**Review by svenaas — APPROVED** [review]
LGTM

**Pull request merged**